pkg/kubelet/pluginmanager/operationexecutor/operation_generator.go

```diff
@@ -135,33 +141,51 @@ func (og *operationGenerator) GenerateRegisterPluginFunc(
 func (og *operationGenerator) GenerateUnregisterPluginFunc(
 	socketPath string,
 	pluginHandlers map[string]cache.PluginHandler,
+	pathToHandlers map[string]cache.HandlerWithName,
 	actualStateOfWorldUpdater ActualStateOfWorldUpdater) func() error {
 
 	unregisterPluginFunc := func() error {
-		client, conn, err := dial(socketPath, dialTimeoutDuration)
-		if err != nil {
-			return fmt.Errorf("UnregisterPlugin error -- dial failed at socket %s, err: %v", socketPath, err)
+		var pair cache.HandlerWithName
+		var handler cache.PluginHandler
+		ok := false
+		if len(pathToHandlers) != 0 {
+			pair, ok = pathToHandlers[socketPath]
 		}
-		defer conn.Close()
+		if !ok {
+			client, conn, err := dial(socketPath, dialTimeoutDuration)
```

Comment:
I don't understand why we need retain dial operation. Unregister operation is triggered by socket file deletion

Reply:
With this PR, the dial is not needed - after the fix is verified.

Reply:
And what the scenario will use this dail operation?